stream.js

```diff
@@ -58,8 +58,23 @@ function WebSocketStream(target, protocols, options) {
 
   proxy.on('close', destroy)
 
+  // for server only
+  // The value of the arguemnt called 'binary' is decided on a chunk by chunk basis by default.
+  // socket.send() modify the passed in options argument.
+  // Hence, reusing the same object on each send call results in a situation, where
+  // any unpassed paramters will be set based on the very first chunk; To overcome this
+  // a new object literal is needed for each invocation of send
```

Comment:
My aim was to ensure that ws.send() will be called with a brand new object, not with an object reference.
So I created the sendOptions helper. And the reason why a new object is needed, is because ws.send() mutates the fields of the options argument based on its own internal logic, but only if those fields are undefined.

So the scenario what I want to avoid is this:
```javascript
// a single options object
var options = { compress: true } // just picked property 'compress' randomly for illustration

socketWriteNode(buffer, 'buffer', next)
// -> options = {binary: true, mask: false, fin:. true, compress: true}
// from now on the above values will be used for all subsequent ws.send() call

socketWriteNode('hello', 'UTF-8', next)
// Note, that the options variable did not specify the frame type,
// so by default, ws.send() will figure it out based on the encoding of the chunk,
// but the first call to socketWriteNode() will set binary to true for every subsequent ws.send() call
// hence 'hello' will be sent in binary date frames too
```

Sorry for the confusing comment, I hope I did manage to clarify it now.